corehq/apps/accounting/management/commands/list_customer_billing_account_software_plan.py

```diff
@@ -0,0 +1,98 @@
+from django.core.management.base import BaseCommand
+from corehq.apps.accounting.models import BillingAccount, Subscription, SubscriptionType
+import csv
+
+
+class Command(BaseCommand):
+    # Always run this command without '--update' first
+    # let Ops review the output before running with '--update'
+    help = 'List customer billing accounts and associated software plans'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--update', action='store_true', default=False,
+                            help='Update all software plans (type PRODUCT or IMPLEMENTATION) '
+                            'under each billing account to the main billing subscription')
+
+    def handle(self, *args, **kwargs):
+        update = kwargs['update']
+        billing_account_names = []
+        while True:
+            account_name = input("Enter a customer billing account name, "
+                                 "type 'ALL' for all customer billing account, "
+                                 "type 'DONE' when finished:\n")
+            if account_name.upper() == 'ALL':
+                billing_account_names = ['ALL']
+                break
+            elif account_name.upper() == 'DONE':
+                break
+            else:
+                try:
+                    BillingAccount.objects.get(name=account_name, is_active=True, is_customer_billing_account=True)
+                except BillingAccount.DoesNotExist:
+                    print(f"Customer billing account {account_name} does not exist")
+                else:
+                    billing_account_names.append(account_name)
+
+        if 'ALL' in billing_account_names:
+            customer_billing_account = BillingAccount.objects.filter(
+                is_customer_billing_account=True, is_active=True)
+        else:
+            customer_billing_account = BillingAccount.objects.filter(
+                is_customer_billing_account=True, is_active=True, name__in=billing_account_names)
+
+        file_path = "/tmp/customer_billing_plans.csv"
+        with open(file_path, 'w', newline='') as csvfile:
+            csvwriter = csv.writer(csvfile)
+            headers = ["Account", "Domain", "Software Plan", "Type", "Do not invoice",
+                       "Start Date", "End Date", "Main Subscription", "Need Update", "Status"]
+            csvwriter.writerow(headers)
+
+            for account in customer_billing_account:
+                subscriptions = Subscription.visible_objects.filter(account=account, is_active=True)
+                if len(subscriptions) == 0:
+                    csvwriter.writerow([account.name, "", "", "", "", "", "", "", "", ""])
+                else:
+                    # Value for csv column Main Subscription
+                    csv_main_subscription = ""
+                    try:
+                        main_subscription = subscriptions.filter(do_not_invoice=False).get()
```

Comment:
what happens in the case that this returns more than one subscription? There are some `BillingAccounts` that have this situation

Reply:
nevermind, I see the following except